pkg/sink/sink.go

```diff
@@ -136,3 +167,96 @@ func createRequestURI(apiVersion, namePlural, namespace string) string {
 	uri = path.Join(uri, namePlural)
 	return uri
 }
+
+func (r Resource) validateEvent(triggerValidate *triggersv1.TriggerValidate, headers http.Header, payload []byte) error {
+	tr, err := r.createValidateTask(triggerValidate, headers, payload)
+	if err != nil {
+		return nil
+	}
+
+	tr, err = r.PipelineClient.TektonV1alpha1().TaskRuns(r.EventListenerNamespace).Create(tr)
+	if err != nil {
+		return err
+	}
+
+	for i := taskRunPollingRetries; i >= 0; i-- {
```

Comment:
As an improvement, we could probably use a watcher here, which has a deadline rather than polling.

Reply:
Yes. I would do that along with a way to configure timeout in a separate pr.